Use elem.srcObject if available. Fallback to elem.src

DiegoRBaquero opened the pull request:
Fixes https://github.com/webtorrent/webtorrent/issues/1556

DiegoRBaquero committed:
Use elem.srcObject if available. Fallback to elem.src

Fixes https://github.com/webtorrent/webtorrent/issues/1556

DiegoRBaquero committed:
Standard fix

feross commented:
Pretty sure that `srcObject` is the new standard. Chrome release notes say this:

" The URL.createObjectURL() method has been removed from the MediaStream interface. This method has been deprecated in 2013 and superseded by assigning streams to HTMLMediaElement.srcObject. The old method was removed because it is less safe, requiring a call to URL.revokeOjbectURL() to end the stream. Other user agents have either deprecated (Firefox) or removed (Safari) this feature feature."

feross commented:
I think we can probably remove the fallback.

feross committed:
Remove fallback to createObjectURL()

We can re-add it if it breaks something, but I'm 99% sure that all browsers support the new method at this point.

feross commented:
Oh @jhiesey rightly points out that this deprecation is only for WebRTC media streams, like video/audio from the `getUserMedia()` API which is not used by WebTorrent. We shouldn't need to change anything here, I think.
